kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/codec/KafkaRequestDecoder.java

```diff
@@ -144,7 +144,7 @@ static short readAcks(ByteBuf in, int startOfMessage, short apiKey, short apiVer
             if (apiVersion < 9) { // Last non-flexible version
                 transactionIdLength = in.readShort();
             }
-            else if (apiVersion == 9 || apiVersion == 10) { // Flexible versions
+            else if (apiVersion <= 11) { // Flexible versions
```

Comment:
I wonder if we should drop the version check here and rely on the unit tests to check we can extract acks successfully. The test already runs through all known api versions.

Reply:
certainly feels a bit funky to need to make a code change here every time the API version changes.

Reply:
Raused as #1389